File: python_converted/src/core/interfaces/types/npc/DialogueManager.py
from typing import Any, List


class DialogueResponse:
    message: str
    tone: str
    emotionalState: str
    subtext?: str
    followUp?: List[str]
class DialogueManager:
    generateDialogue(
    npc: Any,
    context: Any
  ): Awaitable[DialogueResponse>
    generateResponse(
    npc: Any,
    input: str,
    context: Any
  ): Awaitable[DialogueResponse>
    generateEmotionalResponse(
    npc: Any,
    emotion: str,
    intensity: float,
    context: Any
  ): Awaitable[DialogueResponse>
    generateNegotiationResponse(
    npc: Any,
    offerQuality: float,
    context: Any
  ): Awaitable[DialogueResponse>
    generateDeceptionResponse(
    npc: Any,
    detected: bool,
    context: Any
  ): Awaitable[DialogueResponse>
    generateCooperationResponse(
    npc: Any,
    proposal: Any,
    context: Any
  ): Awaitable[DialogueResponse>
    generateCompetitionResponse(
    npc: Any,
    challenge: Any,
    context: Any
  ): Awaitable[DialogueResponse> 

File: src/core/interfaces/types/npc/DialogueManager.ts
import { NPCData } from './npc';
import { InteractionContext } from '../../systems/npc/InteractionSystem';

export interface DialogueResponse {
  message: string;
  tone: string;
  emotionalState: string;
  subtext?: string;
  followUp?: string[];
}

export interface DialogueManager {
  generateDialogue(
    npc: any,
    context: any
  ): Promise<DialogueResponse>;
  
  generateResponse(
    npc: any,
    input: string,
    context: any
  ): Promise<DialogueResponse>;
  
  generateEmotionalResponse(
    npc: any,
    emotion: string,
    intensity: number,
    context: any
  ): Promise<DialogueResponse>;
  
  generateNegotiationResponse(
    npc: any,
    offerQuality: number,
    context: any
  ): Promise<DialogueResponse>;
  
  generateDeceptionResponse(
    npc: any,
    detected: boolean,
    context: any
  ): Promise<DialogueResponse>;
  
  generateCooperationResponse(
    npc: any,
    proposal: any,
    context: any
  ): Promise<DialogueResponse>;
  
  generateCompetitionResponse(
    npc: any,
    challenge: any,
    context: any
  ): Promise<DialogueResponse>;
} 